### PR TITLE
Ignored bot and system messages in messageWillBePosted hook

### PR DESCRIPTION
**Pull request by dbejanishvili**
#### Summary
Used ShouldProcessMessage to skip posts from bots and system messages.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-github/issues/266

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -162,6 +162,16 @@ func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*mode
 		return nil, ""
 	}
 
+	shouldProcessMessage, e := p.Helpers.ShouldProcessMessage(post)
```
Comment (hanzei) [inline server/plugin.go:165]: Nit pick: Using `e` as a name for an `error` is quite uncommon. I would suggest using `error` and renaming the `*model.Apperror` in L176 into `appError`.

Comment (dbejanishvili) [reply server/plugin.go:165]: yeah, I thought that would be better too, I just tried not to change any outside code

**Comment by DHaussermann** [comment]
@larkox or @hanzei, If you have a bit of time, could one of you peer-test this PR using the  peer testing process documented here https://mattermost.atlassian.net/wiki/spaces/INT/pages/619937892/Peer+Testing+Process

**Comment by hanzei** [comment]
@larkox Could you please take this one? I looking into the other two.

**Comment by larkox** [comment]
Any message posted by a user not registered to github would also throw the error. Do we want to avoid that?

Also, I am not sure about the privacy of this feature. Since the poster token is used, potentially it may show to anyone content of private repositories. Do we want to avoid this here too? Or is this handled in a different part of the code?

**Comment by larkox** [comment]
@hanzei :point_up:

**Comment by hanzei** [comment]
> Any message posted by a user not registered to github would also throw the error. Do we want to avoid that?

While that is not a new behaviour, it defiantly spammy. An error should only be logged if `getGitHubUserInfo` returned an error different to `apiErrorIDNotConnected`. @dbejanishvili Would you be interested in also fixing that issue in this PR?

> Also, I am not sure about the privacy of this feature. Since the poster token is used, potentially it may show to anyone content of private repositories. Do we want to avoid this here too? Or is this handled in a different part of the code?

@larkox That is a very valid concern, but out of scope of this PR. Would you mind opening a new issue for that? We probably need also input from the security team.

**Comment by dbejanishvili** [comment]
Sure

**Comment by hanzei** [comment]
@larkox This is ready for another round of QA review

**Pull request merged**